src/librustc_mir/build/expr/as_rvalue.rs

```diff
@@ -125,35 +125,36 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                 //     to the same MIR as `let x = ();`.
 
                 // first process the set of fields
-                let fields: Vec<_> =
-                    fields.into_iter()
-                          .map(|f| unpack!(block = this.as_operand(block, f)))
-                          .collect();
+                let fields: Vec<_> = fields.into_iter()
+                                           .map(|f| unpack!(block = this.as_operand(block, f)))
+                                           .collect();
 
                 block.and(Rvalue::Aggregate(AggregateKind::Vec, fields))
             }
             ExprKind::Tuple { fields } => { // see (*) above
                 // first process the set of fields
-                let fields: Vec<_> =
-                    fields.into_iter()
-                          .map(|f| unpack!(block = this.as_operand(block, f)))
-                          .collect();
+                let fields: Vec<_> = fields.into_iter()
+                                           .map(|f| unpack!(block = this.as_operand(block, f)))
+                                           .collect();
 
                 block.and(Rvalue::Aggregate(AggregateKind::Tuple, fields))
             }
             ExprKind::Closure { closure_id, substs, upvars } => { // see (*) above
-                let upvars =
-                    upvars.into_iter()
-                          .map(|upvar| unpack!(block = this.as_operand(block, upvar)))
-                          .collect();
+                let upvars = upvars.into_iter()
+                                   .map(|upvar| unpack!(block = this.as_operand(block, upvar)))
+                                   .collect();
                 block.and(Rvalue::Aggregate(AggregateKind::Closure(closure_id, substs), upvars))
             }
             ExprKind::Adt { adt_def, variant_index, substs, fields, base } => { // see (*) above
                 // first process the set of fields
-                let fields_map: FnvHashMap<_, _> =
-                    fields.into_iter()
-                          .map(|f| (f.name, unpack!(block = this.as_operand(block, f.expr))))
-                          .collect();
+                let fields_map: FnvHashMap<_, _> = fields.into_iter()
```

Comment:
@marcusklaas 

> One of the more simple heuristics we could start with is comparing the result of formatting on the same line and breaking after the assignment and choosing the one with the fewest line breaks (no break on tie).

@nrc 

> the majority of times I see this is in a let statement, so it would probably get us 90% of the way there

yes to both